### PR TITLE
deploy button

### PR DESCRIPTION
**Pull request by grosser**
@zendesk/samson 

can try it here ...
https://dashboard.heroku.com/new?button-url=https%3A%2F%2Fgithub.com%2Fzendesk%2Fsamson%2Fblob%2Fgrosser%2Fbutton%2FREADME.md&template=https%3A%2F%2Fgithub.com%2Fzendesk%2Fsamson%2Fblob%2Fgrosser%2Fbutton%2FREADME.md

(had to push app.json to master for it to work ... because heroku does not support branches with `/`

readme preview:
https://github.com/zendesk/samson/blob/grosser/button/README.md

demo: https://samson-demo.herokuapp.com/

### CODE REVIEW

**Comment by grosser** [comment]
got a bit further now ... app boots fine, but login is still broken ...

File: config/initializers/session_store.rb [hunk 1]
```diff
@@ -1,3 +1,11 @@
...
-Samson::Application.config.session_store :cookie_store, key: '_samson_session', domain: :all
+config = Samson::Application.config
+options = {key: '_samson_session'}
+
+# when using multiple domains we have to share our cookies
+# this breaks session on heroku when using *.herokuapp.com
+used_domains = [config.samson.deploy_origin, config.samson.stream_origin, config.samson.uri.to_s]
+options[:domain] = :all if used_domains.uniq.size != 1
+
+Samson::Application.config.session_store :cookie_store, options
```
Comment (grosser) [inline config/initializers/session_store.rb]: FYI @steved

**Comment by grosser** [comment]
@zendesk/samson ready for review!

**Comment by grosser** [comment]
using `puma -C` is recommended by heroku, but ends up always having database locking issues ... `rails s` somehow works ... even when both use the same thread counts

**Comment by grosser** [comment]
review plz @steved @zendesk/samson

**Comment by jonmoter** [comment]
:+1:

**Pull request merged**